dev_guide/builds.adoc

```diff
@@ -519,9 +519,52 @@ customStrategy:
 ----
 ====
 
+[[build-inputs]]
+== Build Inputs
+
+There are several ways to provide content for builds to operate on. In order of precedence:
+
+* Inline Dockerfile definitions
+* Content extracted from existing images
+* Git repositories
+* Binary inputs
+
+These can be combined into a single build. As the inline Dockerfile takes precedence, it can overwrite any other file named Dockerfile provided by another input. Binary input and Git repository are mutually exclusive inputs.
+
+When the build is run, a working directory is constructed and all input content is placed in the working directory (e.g. the input git repository is cloned into the working directory, files specified from input images are copied into the working directory using the target path).  Next the build process will `*cd*` into the `*contextDir*` if one is defined.  Then the inline `*Dockerfile*` (if any) is written to the current directory.  Last, the content from the current directory will be provided to the build process for reference by the `*Dockerfile*`, `*assemble*` script, or custom builder logic.  This means any input content that resides outside the `*contextDir*` will be ignored by the build.
```

Comment:
I think this could work as an ordered list instead of having to dig through a big paragraph?